cfme/tests/ansible/test_embedded_ansible_automate.py

```diff
@@ -419,3 +421,56 @@ def test_variable_pass(request, appliance, setup_ansible_repository, import_data
         provision_request = appliance.collections.requests.instantiate(request_description)
         provision_request.wait_for_request(method="ui")
         request.addfinalizer(provision_request.remove_request)
+
+
+@pytest.mark.tier(1)
+@pytest.mark.ignore_stream("5.10")
+@pytest.mark.meta(automates=[1677575])
+@pytest.mark.parametrize(
+    "import_data",
+    [DatastoreImport("bz_1677575.zip", "bz_1677575", None)],
+    ids=["datastore"],
+)
+def test_import_domain_containing_playbook_method(request, appliance, setup_ansible_repository,
+                                                  import_data):
+    """This test case tests support of Export/Import of Domain with Ansible Method
+
+    Bugzilla:
+        1677575
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseimportance: high
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.11
+        casecomponent: Automate
+        tags: automate
+        setup:
+            1. Add playbook repository
+            2. Create a new automate method with playbook type.
+            3. Fill the required fields, for instance repository and playbook.
+            4. Export this datastore.
+            5. Playbook method fields are stored as a names instead of IDs. (this is not
+               possible via automate need to check manually in method yaml)
+        testSteps:
+            1. Import the exported datastore.
+            2. Change name of playbook in method yaml to invalid playbook name
+        expectedResults:
+            1.
+            2. Proper error should be displayed while importing datastore with invalid playbook
+    """
+    # Download datastore file from FTP server
+    fs = FTPClientWrapper(cfme_data.ftpserver.entities.datastores)
+    file_path = fs.download(import_data.file_name)
+
+    # Import datastore file to appliance
+    datastore = appliance.collections.automate_import_exports.instantiate(
+        import_type="file", file_path=file_path
+    )
+    domain = datastore.import_domain_from(import_data.from_domain, import_data.to_domain)
+    request.addfinalizer(domain.delete_if_exists)
+    view = appliance.browser.create_view(FileImportSelectorView)
+    error_msg = "Playbook 'invalid_1677575.yml' not found in repository 'test_playbooks_automate'"
```

Comment:
Is there a way to avoid hardcoding repository name ?

Reply:
@sshveta 
Yeah. I agree over here. But while importing datastore; we are not able to change repository name from datastore yamls. Hence need to have repository, which is there in datastore. As I commented:
```# Repository name(test_playbooks_automate) is static because this is already in the datastore
    # imported. If we used(fauxfactory.gen_alpha()) here then datastore import will fail with no
    # ansible repository available.```

Reply:
@ganeshhubale : Try reading the datastore file in this way 
https://github.com/ManageIQ/integration_tests/blob/master/cfme/fixtures/service_fixtures.py#L253

If that doesn't work please add a comment  that this repository name is coming from the datastore edited manually .